WalletWasabi/Models/SmartTransaction.cs

```diff
@@ -124,6 +126,115 @@ public static IComparer<SmartTransaction> GetBlockchainComparer()
 			});
 		}
 
+		#region LineSerialization
+
+		public string ToLine()
+		{
+			// GetHash is also serialized, so file can be interpreted with our eyes better.
+
+			return string.Join(':',
+				GetHash(),
+				Transaction.ToHex(),
+				Height,
+				BlockHash,
+				BlockIndex,
+				Label,
+				FirstSeen.ToUnixTimeSeconds(),
+				IsReplacement);
+		}
+
+		public static SmartTransaction FromLine(string line, Network expectedNetwork)
+		{
+			line = Guard.NotNullOrEmptyOrWhitespace(nameof(line), line, trim: true);
+			expectedNetwork = Guard.NotNull(nameof(expectedNetwork), expectedNetwork);
+
+			var parts = line.Split(':', StringSplitOptions.None).Select(x => x.Trim()).ToArray();
+
+			// Find the Transaction hex, it must be always present.
+			(Transaction transaction, int transactionHexIndex) = ParseTransactionHex(expectedNetwork, parts);
+
+			try
+			{
+				// First is redundand txhash serialization.
+				var heightString = parts[transactionHexIndex + 1];
+				var blockHashString = parts[transactionHexIndex + 2];
+				var blockIndexString = parts[transactionHexIndex + 3];
+				var label = parts[transactionHexIndex + 4];
+				var firstSeen = parts[transactionHexIndex + 5];
+				var isReplacementString = parts[transactionHexIndex + 6];
+
+				if (!Height.TryParse(heightString, out Height h))
+				{
+					h = Height.Unknown;
+				}
+				if (!uint256.TryParse(blockHashString, out uint256 bh))
+				{
+					bh = null;
+				}
+				if (!int.TryParse(blockIndexString, out int bi))
+				{
+					bi = 0;
+				}
+				var sl = new SmartLabel(label);
+				DateTimeOffset fs = default;
+				if (long.TryParse(firstSeen, out long unixSeconds))
+				{
+					fs = DateTimeOffset.FromUnixTimeSeconds(unixSeconds);
+				}
+				if (!bool.TryParse(isReplacementString, out bool ir))
+				{
+					ir = false;
+				}
+
+				return new SmartTransaction(transaction, h, bh, bi, sl, ir, fs);
+			}
+			catch (Exception ex)
+			{
+				Logger.LogDebug(ex);
+				return new SmartTransaction(transaction, Height.Unknown);
+			}
+		}
+
+		private static (Transaction tx, int txHexIndex) ParseTransactionHex(Network expectedNetwork, string[] parts)
+		{
+			Transaction tx = null;
+			int txHexIndex = 1;
+			try
+			{
+				tx = Transaction.Parse(parts[txHexIndex], expectedNetwork);
+			}
+			catch
+			{
+				for (txHexIndex = 0; txHexIndex < parts.Length; txHexIndex++)
```

Comment:
How is this possible? We have a strict order at ToLine() and also StringSplitOptions.None is used on Split. IMO it is dangerous to let this assumption.

Reply:
Yes, I have the same question. It seems the design assumes that file could be manually modified by someone and write things in a different order and do some other modifications like adding spaces (that's why the `Trim`) and things like that. Why is this?

Reply:
The idea was so to SmartTransaction can be created from a txhex alone (just like a transaction can be.) But I ruined it for you guys: https://github.com/zkSNACKs/WalletWasabi/pull/2216/commits/d0051ec2b239d9124b7bb85439ac06daecef6dd5